### PR TITLE
updateKey() should throws if not TLS Handshake

### PR DESCRIPTION
**Pull request by juanarbol**
##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

**Note:** this test is unstable, sometimes it's passes, sometimes doesn't, seems that handshake is completed before excepcion is handled, I've tried `setInmediate` and `process.nextTick`, did not work; so I'd love to receive some suggestions or opinions.

### CODE REVIEW

**Comment by jasnell** [comment]
The updateKey part of this is going to be moved to it's own separate test very soon

**Comment by juanarbol** [comment]
I get it, how could I help with this?

**Comment by jasnell** [comment]
If you'd like to try putting together that separate test, that would be excellent :-)

**Comment by juanarbol** [comment]
Ok, I will create a new file, named as `test-quic-updatekey,js`, is there something else more than this exception to be tested, to be written on this file?

**Comment by jasnell** [comment]
Heads up, this will need to be rebased

File: test/parallel/test-quic-updatekey.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  cert,
+  ca,
+  requestCert: true,
+  rejectUnauthorized: false,
+  alpn: kALPN,
+  maxCryptoBuffer: 4096,
+});
+server.on('session', common.mustCall((session) => {
+  process.nextTick(() => {
+    assert.throws(() => session.updateKey(), {
```
Comment (jasnell) [inline test/parallel/test-quic-updatekey.js]: So, investigating further... the issue around updateKey is that a key update can only be triggered (a) only after the TLS handshake has been confirmed and (b) while we're not in the middle of another key update. The TLS handshake is confirmed as soon as we receive the first 1RTT bytes from the peer but ngtcp2 does not really give us a notification at that point so we're going to have to guess somewhat. I'll be working through the semantics on that and will be implementing a `'handshakeConfirmed'` type of event that will flip the switch that enables `updateKey()` to work. Expect that within the next week I hope

Comment (juanarbol) [reply test/parallel/test-quic-updatekey.js]: Sounds good @jasnell, tell me if you need some help.

**Comment by juanarbol** [comment]
Closing

**Pull request closed without merging**